test/test_integration.py

```diff
@@ -215,6 +215,69 @@ def test_upload_gallery(self):
 
         self.snapshot('test_upload_gallery')
 
+    def test_loading_slider(self):
+        app = dash.Dash(__name__)
+
+        app.layout = html.Div([
+            html.Label(id='test-div', children=['Horizontal Slider']),
+            dcc.Slider(
+                id='horizontal-slider',
+                min=0,
+                max=9,
+                marks={i: 'Label {}'.format(i) if i == 1 else str(i)
+                       for i in range(1, 6)},
+                value=5,
+            ),
+        ])
+
+        @app.callback(
+            Output('horizontal-slider', 'value'),
+            [Input('test-div', 'children')]
+        )
+        def delayed_value(children):
+            time.sleep(5)
```

Comment:
I'd like to get away from this pattern of adding super long sleeps. Can we perhaps use a [`Condition`](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Condition) or something, so you wait until the loading state is seen, then tell the callback it's OK to return?